packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.scss

```diff
@@ -17,6 +21,7 @@ $resizerColor: $ms-color-neutralTertiaryAlt;
   border-bottom: 1px solid $dividerColor;
   cursor: default;
   user-select: none;
+  z-index: $ms-zIndex-front;
```

Comment:
Adding picture.  I have a z-index in for the non-header rows.  The image below is without the z-index on the header row (as well as blue border, to show the effect).  The top row in the table is cover the bottom border of the header cell without a z-index.
![screen shot 2017-05-18 at 11 07 32 am](https://cloud.githubusercontent.com/assets/1291968/26216821/5057ec50-3bba-11e7-9bdc-759d65e7f510.png)